b2g/chrome/content/remote_control.js

```diff
@@ -20,57 +20,99 @@ const kPrefPresentationDiscoverable = "dom.presentation.discoverable";
+      debug("observe: " + topic + ", " + data);
 
-  var networkStatusMonitor = {
-    // nsIObserver
-    observe: function(subject, topic, data) {
       switch (topic) {
         case "network-active-changed": {
           if (!subject) {
             // Stop service when there is no active network
-            remoteControlScope.RemoteControlService.stop();
+            this.stopService();
             break;
           }
 
           // Start service when active network change with new IP address
           // Other case will be handled by "network:offline-status-changed"
-          if (!Services.io.offline && Services.prefs.getBoolPref(kPrefPresentationDiscoverable)) {
-            importRemoteControlService();
-            remoteControlScope.RemoteControlService.start();
+          if (!Services.io.offline) {
+            this.startService();
           }
+
           break;
         }
         case "network:offline-status-changed": {
           if (data == "offline") {
             // Stop service when network status change to offline
-            remoteControlScope.RemoteControlService.stop();
-          } else if (Services.prefs.getBoolPref(kPrefPresentationDiscoverable)) {
+            this.stopService();
+          } else { // online
             // Resume service when network status change to online
-            importRemoteControlService();
-            remoteControlScope.RemoteControlService.start();
+            this.startService();
           }
           break;
         }
+        case "nsPref:changed": {
+          if (data != kPrefPresentationDiscoverable) {
+            break;
+          }
+
+          if (!this.startService()) {
+            this.stopService();
+          }
+
+          break;
+        }
         default:
           break;
       }
-    }
-  };
+    },
 
-  if (Ci.nsINetworkManager) {
-    Services.obs.addObserver(networkStatusMonitor, "network-active-changed", false);
-    Services.obs.addObserver(networkStatusMonitor, "network:offline-status-changed", false);
-  }
+    onTimeChange: function sm_onTimeChange() {
+      debug("onmoztimechange");
+      // Notice remote-control service the time is changed
+      remoteControlScope.RemoteControlService &&
+      remoteControlScope.RemoteControlService.onTimeChange(getTime());
+    },
+
+    startService: function sm_startService() {
+      debug("startService");
+      // Start only when presentation is discoverable
+      if (this.discoverable) {
+        importRemoteControlService();
+        remoteControlScope.RemoteControlService.start();
+        return true;
+      }
+      return false;
+    },
+
+    stopService: function sm_stopService() {
+      debug("stopService");
+      remoteControlScope.RemoteControlService &&
+      remoteControlScope.RemoteControlService.stop();
+    },
+
+    get discoverable() {
+      return Services.prefs.getBoolPref(kPrefPresentationDiscoverable);
+    },
+  };
 
-  setupRemoteControl();
+  statusMonitor.init();
```

Comment:
If the time-change event is fired before network is connected, then ```onTimeChange``` will be the entry point for the service, otherwise, the service will be started upon network is alive, so I think it's not specific for nexus player. 

I'll check  on mulet and see what happens.